components/input/textFieldv2/TextFieldv2.jsx

```diff
@@ -0,0 +1,383 @@
+ *
+ * @param {object} - An object containing various properties to configure the TextField component.
+ * @param {string} id - The id of the input field.
+ * @param {string} name - The name of the input field.
+ * @param {string|number} label - The label text for the input field.
+ * @param {string|number} placeholder - The placeholder text for the input field.
+ * @param {string} type - The type of the input field (text, email, password, number, textarea).
+ * @param {string|number} value - The value of the input field.
+ * @param {boolean} required - Whether the input field is required.
+ * @param {string} defaultValue - The default value for the input field.
+ * @param {function} onFocus - The event handler for the input field's focus event.
+ * @param {function} onBlur - The event handler for the input field's blur event.
+ * @param {function} onChange - The event handler for the input field's change event.
+ * @param {string} size - The size of the input field (sm, md, lg).
+ * @param {string} border - The border style of the input field (default, bottom, none).
+ * @param {string} theme - The theme of the input field (light, dark).
+ * @param {ReactNode} LeftComponent - The component to render on the left side of the input field.
+ * @param {ReactNode} RightComponent - The component to render on the right side of the input field.
+ * @param {string} className - The CSS class name for the input field.
+ * @param {boolean} disabled - Whether the input field is disabled.
+ * @param {object} input- Additional to pass to the input field.
+ * @param {object} feedback - The feedback message for the input field.
+ * @param {object} count - The character count configuration for the input field.
+ * @param {boolean} feedbackAndCount - Whether to display both feedback and character count.
+ * @param {number} maxLength - The maximum length of the input field.
+ * @param {function} onKeyDown - The event handler for the input field's key down event.
+ * @param {boolean} autocomplete - Whether to enable autocomplete for the input field.
+ * @param {object} autocompleteOptions - The options for the autocomplete feature.
+ * @param {boolean} custom - Whether to use custom styling for the input field.
+ * @returns {ReactNode} The rendered TextField component.
+ */
+const TextField = forwardRef((props, inputRef) => {
+	const {
+		id,
+		name,
+		label,
+		placeholder,
+		type,
+		value,
+		required,
+		defaultValue,
+		onFocus,
+		onBlur,
+		onChange,
+		size,
+		border,
+		theme,
+		LeftComponent,
+		RightComponent,
+		className,
+		disabled,
+		inputProps,
+		feedback,
+		count,
+		feedbackAndCount,
+		maxLength,
+		onKeyDown,
+		autocomplete,
+		autocompleteOptions,
+		custom,
+	} = props;
+
+	const { current: isControlled } = useRef(value !== undefined);
+
+	// for uncontrolled input
+	const [uncontrolledValue, setUncontrolledValue] = useState(defaultValue);
+	const [inputType, setInputType] = useState('textarea');
```

Comment:
Here in the initialValue, you can rewrite this as below

`useState(type || 'textarea')`

or

```
useState(()=>{
   if(type){
       return type
   }
   return "textArea"
})
```